internal/pkg/nats/client.go

```diff
@@ -142,10 +158,48 @@ func (c *Client) Subscribe(topics []types.TopicChannel, messageErrors chan error
 		if err != nil {
 			return err
 		}
+
+		c.existingSubscriptions[tc.Topic] = subscription
 	}
+
 	return nil
 }
 
+// Request publishes a request and waits for a response
+func (c *Client) Request(message types.MessageEnvelope, topic string, timeout time.Duration) (*types.MessageEnvelope, error) {
+	return pkg.DoRequest(c.Subscribe, c.Unsubscribe, c.Publish, message, topic, c.config.ResponseTopicPrefix, timeout)
+}
+
+// Unsubscribe to unsubscribe from the specified topics.
+func (c *Client) Unsubscribe(topics ...string) error {
+	if c.connection == nil {
+		return fmt.Errorf("cannot unsubscribe with disconnected client")
+	}
+
+	c.subscriptionMutex.Lock()
+	defer c.subscriptionMutex.Unlock()
+
+	var errs error
+	for _, topic := range topics {
+		subscription, exists := c.existingSubscriptions[topic]
+		if !exists {
+			continue
+		}
+
+		if subscription != nil {
+			err := subscription.Unsubscribe()
+			if err != nil {
+				errs = multierror.Append(errs, fmt.Errorf("unable to unsubscribe to topic '%s': %v", topic, err))
+				continue
+			}
+		}
```

Comment:
i suggest we logging the else case- it is sort of corner case right? (existing subscription but subscription is nil or empty?)

Reply:
Module doesn't log messages, just return errors. The point here is if the subscription doesn't exist, not need  and can't unsubscribe. I'll add that as a comment.